holoviews/element/raster.py

```diff
@@ -274,9 +279,10 @@ def __init__(self, data, kdims=None, vdims=None, bounds=None, extents=None,
 
         # Ensure coordinates are regularly sampled
 
-        rtol = config.image_rtol if rtol is None else rtol
-        validate_regular_sampling(self, 0, rtol)
-        validate_regular_sampling(self, 1, rtol)
+        self.set_param(rtol=config.image_rtol) if rtol is None else self.set_param(rtol=rtol)
```

Comment:
Rather than using ``set_param`` here, the ``rtol`` value should be passed to the Dataset constructor above, something like:

```
rtol = (config.image_rtol if self.rtol is None else self.rtol) if rtol is None else rtol
Dataset.__init__(self, data, kdims=kdims, vdims=vdims, extents=extents, **params)
```

and then change the default ``rtol`` to None.

Reply:
Agreed.

Reply:
I don't quite understand and I am probably missing something
if `set_param` is not used then `self.rtol = None` always, so in the conditional statement the `config.image_rtol if self.rtol is None`  is always True and we never hit the ` else self.rtol `

Reply:
Further, unless `rtol` is a `param` the clone method in holoviews/core/dimension.py wont copy it and I get the following issue, which is what caused me to make rtol a param in.
```
  File "my_path/holoviews/core/dimension.py", line 572, in clone
    return clone_type(data, *args, **{k:v for k,v in settings.items()
```

Reply:
>if set_param is not used then self.rtol = None always, so in the conditional statement the config.image_rtol if self.rtol is None is always True and we never hit the else self.rtol

You can override parameters at the class level, e.g. if you do ``hv.Image.rtol = 0.000001`` then ``self.rtol`` will not be None.

>Further, unless rtol is a param the clone method in holoviews/core/dimension.py wont copy it and I get the following issue, which is what caused me to make rtol a param in the first place.

This should not be the case but Image is a bit special in terms of multiple inheritance so I'd have to look more closely.